src/components/AbstractComponent/index.js

```diff
@@ -0,0 +1,13 @@
+export class AbstractComponent {
+    constructor(props = undefined) {
+        if (typeof props !== "object" || props.constructor !== Object) {
+            props = {};
+        }
```

Comment:
А зачем по дефолту undefined?
```suggestion
    constructor(props = {}) {
        if (typeof props !== "object" || props.constructor !== Object) {
            props = {};
        }
```